hyperspy/axes.py

```diff
@@ -655,6 +720,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_index_changed(self):
+        self._update_attributes()
+        self.events.indices_changed.trigger(axes_manager=self)
+
+    def _on_view_changed(self):
+        self._update_attributes()
+        self.events.axes_changed.trigger()
```

Comment:
Several things trigger `axes_changed` - view change is just one of them - so I don't believe it would be correct for them to correspond directly. Now there might be an additional event called `view_changed`, but i haven't yet started implementing "future" events, i.e. events that might be useful some day. I'm also not sure if those events belong in this PR?